src/Pyro4/util.py

```diff
@@ -369,7 +369,16 @@ def dict_to_class(cls, data):
                 exceptiontype = getattr(sqlite3, short_classname)
                 if issubclass(exceptiontype, BaseException):
                     return SerializerBase.make_exception(exceptiontype, data)
-
+        
+        # serpent serializes float('nan') as {'__class__':'float','value':'nan'}. 
+        #  I'm not sure if this is the most elegant way to handle that edge case 
+        #  (it'd likely make more sense for this knowledge to live in 
+        #  SerpentSerializer?) - but it does the trick. ~ jackjweinstein 1/26/17

```

Comment:
Notice that serpent itself cannot handle this, by design.  It's based around Python literal expressions, and the deserialization is done not by the serpent library itself but by the standard library's ast.literal_eval().  You can't teach that thing to do custom type conversions and I think that is a good thing because the whole idea is to never instantiate custom classes.  Unfortunately we're dealing with a built in type here (float) so it is safe, but yeah.